python_modules/dagster/dagster/_core/definitions/asset_reconciliation_sensor.py

```diff
@@ -366,9 +388,7 @@ def determine_asset_partitions_to_reconcile(
                         # if they don't have the same partitioning, then we can't launch a run that
                         # targets both, so we need to wait until the parent is reconciled before
                         # launching a run for the child
-                        and asset_graph.have_same_partitioning(
-                            parent.asset_key, candidate.asset_key
-                        )
+                        and parent.partition_key == candidate.partition_key
```

Comment:
should we keep the original conditional as well? I forget how partition keys are generated but is it possible that a weekly and a daily partitions def both have a key of `2022-01-01`?

Reply:
I think you are right